### ISSUE TITLE
Elequence unable to load on nvda 2021.1 and higher.

### ISSUE THREAD

**Issue by austinpinto** [pumper42nickel/eloquence_threshold#22]
nvda 2021 speech.*Command attributes are gone, replaced by speech.commands.*Command. 
so elequence doesnt start when nvda loads.
this is a part of the log.
speach.ERROR - synthDriverHandler.setSynth (08:11:31.778) - MainThread (13208):
setSynth
Traceback (most recent call last):
  File "synthDriverHandler.pyc", line 448, in setSynth
  File "synthDriverHandler.pyc", line 419, in getSynthInstance
  File "synthDriverHandler.pyc", line 383, in _getSynthDriver
  File "importlib\__init__.pyc", line 127, in import_module
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "C:\Users\austi\AppData\Roaming\nvda\addons\Eloquence\synthDrivers\eloquence.py", line 77, in <module>
    class SynthDriver(synthDriverHandler.SynthDriver):
  File "C:\Users\austi\AppData\Roaming\nvda\addons\Eloquence\synthDrivers\eloquence.py", line 80, in SynthDriver
    speech.IndexCommand,
AttributeError: module 'speech' has no attribute 'IndexCommand'
ERROR - synthDriverHandler.getSynthList (08:11:33.854) - MainThread (13208):
Error while importing SynthDriver eloquence
Traceback (most recent call last):
  File "synthDriverHandler.pyc", line 394, in getSynthList
  File "synthDriverHandler.pyc", line 383, in _getSynthDriver
  File "importlib\__init__.pyc", line 127, in import_module
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "C:\Users\austi\AppData\Roaming\nvda\addons\Eloquence\synthDrivers\eloquence.py", line 77, in <module>
    class SynthDriver(synthDriverHandler.SynthDriver):
  File "C:\Users\austi\AppData\Roaming\nvda\addons\Eloquence\synthDrivers\eloquence.py", line 80, in SynthDriver
    speech.IndexCommand,
AttributeError: module 'speech' has no attribute 'IndexCommand'

**Comment by pumper42nickel** [comment]
Thanks for flagging this, let me take a look.

**Comment by pumper42nickel** [comment]
Fixed in  version 0.20210313.01

**Comment by tapper82** [comment]
Hi I installed the new update but eloquence is not showing up in the list of sinths.

**Comment by pumper42nickel** [comment]
Paste your log here.

**Comment by austinpinto** [comment]
if you make a new version and want me to beta test it let me know,
also provide the download link for that new version and i will give
feedback.

On 3/17/21, pumper42nickel ***@***.***> wrote:
> Paste your log here.
>
> On 3/17/21, tapper82 ***@***.***> wrote:
>> Hi I installed the new update but eloquence is not showing up in the list
>> of
>> sinths.
>>
>> --
>> You are receiving this because you modified the open/close state.
>> Reply to this email directly or view it on GitHub:
>> https://github.com/pumper42nickel/eloquence_threshold/issues/22#issuecomment-801094865
>
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/pumper42nickel/eloquence_threshold/issues/22#issuecomment-801199819


-- 
to listen to the Blind Android Users Podcast visit.
www.blindandroidusers.com
To join our mailing list, send those “join requests” to:
***@***.***
You may follow us on twitter using our handle:
@BlindDroidUsers
https://twitter.com/@BlindDroidUsers
We do have a Telegram group that you can join and share stories or
whatever with our members at:
https://t.me/ANATAD

**Comment by kieranlittle667** [comment]
Got this from my log, running latest addon and latest nvda snapshot.
DEBUG - core.main (22:53:22.501) - MainThread (10824):
Initializing speech
DEBUG - addonHandler.Addon.addToPackagePath (22:53:22.501) - MainThread (10824):
Addon Eloquence added to synthDrivers package path
DEBUGWARNING - Python warning (22:53:22.519) - MainThread (10824):
C:\Users\user\AppData\Roaming\nvda\addons\Eloquence\synthDrivers\_eloquence.py:318: SyntaxWarning: assertion is always true, perhaps remove parentheses?
  assert(not temporary, "Can only set vParams permanently from another thread.")
ERROR - synthDriverHandler.setSynth (22:53:22.534) - MainThread (10824):
setSynth
Traceback (most recent call last):
  File "synthDriverHandler.pyc", line 451, in setSynth
  File "synthDriverHandler.pyc", line 422, in getSynthInstance
  File "synthDriverHandler.pyc", line 386, in _getSynthDriver
  File "importlib\__init__.pyc", line 127, in import_module
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "C:\Users\user\AppData\Roaming\nvda\addons\Eloquence\synthDrivers\eloquence.py", line 106, in <module>
    class SynthDriver(synthDriverHandler.SynthDriver):
  File "C:\Users\user\AppData\Roaming\nvda\addons\Eloquence\synthDrivers\eloquence.py", line 107, in SynthDriver
    supportedSettings=(SynthDriver.VoiceSetting(), SynthDriver.VariantSetting(), SynthDriver.RateSetting(), SynthDriver.PitchSetting(),SynthDriver.InflectionSetting(),SynthDriver.VolumeSetting(), driverHandler.NumericDriverSetting("hsz", "Head Size"), driverHandler.NumericDriverSetting("rgh", "Roughness"), driverHandler.NumericDriverSetting("bth", "Breathiness"), driverHandler.BooleanDriverSetting("backquoteVoiceTags","Enable backquote voice &tags", True), driverHandler.BooleanDriverSetting("ABRDICT","Enable &abbreviation dictionary", False), driverHandler.BooleanDriverSetting("phrasePrediction","Enable phrase prediction", False))
AttributeError: module 'driverHandler' has no attribute 'NumericDriverSetting'

Let me know if you need anything more from my log.

**Comment by pumper42nickel** [comment]
Yeah, I've heard that NVDA are refactoring speech submodule, so they
are breaking all synths. I will take a look later when they release
2021.1 stable - as they will keep breaking things before then.
If 2021.1 is out and this is still broken - please ping me here.

**Comment by austinpinto** [comment]
hi.
you can try it with nvda alfa snapshots.
once you install the latest alfa snapshot, they wont brake anything more.

**Comment by tapper82** [comment]
Hi can you fix this pleas?

**Comment by pumper42nickel** [comment]
Try version v0.20210417.01.

**Comment by austinpinto** [comment]
hi. wow now it works with the latest update.
but there is a small problem.
after installing  the latest version of this addon nvda restarts, and
after it restarts the sound changes and it sounds bad.
also when ever nvda restarts the sound changes.
to solve what i must do is press the capslock control and left arrow
and go to the place that it says varrient read.
then press capslock control down arrow once and the sound returns back
to normal.

On 4/18/21, pumper42nickel ***@***.***> wrote:
> Try version v0.20210417.01.
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/pumper42nickel/eloquence_threshold/issues/22#issuecomment-821883694


-- 
to listen to the Blind Android Users Podcast visit.
www.blindandroidusers.com
To join our mailing list, send those “join requests” to:
***@***.***
You may follow us on twitter using our handle:
@BlindDroidUsers
https://twitter.com/@BlindDroidUsers
We do have a Telegram group that you can join and share stories or
whatever with our members at:
https://t.me/ANATAD
